packages/client/src/objectSet/ObjectSetListenerWebsocket.ts

```diff
@@ -436,8 +457,42 @@ export class ObjectSetListenerWebsocket {
     const referenceUpdates = payload.updates.filter((update) =>
       update.type === "reference"
     );
+    const osdkObjectsWithReferenceUpdates = await Promise.all(
+      referenceUpdates.map(async (o) => {
+        const osdkObjectArray = await convertWireToOsdkObjects(
+          this.#client,
+          [{
+            __apiName: o.objectType,
+            __primaryKey: Object.values(o.primaryKey)[0],
+            ...o.primaryKey,
+            [o.property]: o.value,
+          }],
+          undefined,
+        ) as Array<Osdk.Instance<any, never, any>>;
+        const singleOsdkObject = osdkObjectArray[0] ?? undefined;
+        return singleOsdkObject != null
+          ? {
+            object: singleOsdkObject,
+            state: "ADDED_OR_UPDATED" as ObjectState,
+          }
+          : undefined;
+      }),
+    );
+
+    for (const osdkObject of osdkObjectsWithReferenceUpdates) {
+      if (osdkObject != null) {
+        sub.listener.onChange?.(osdkObject);
+      }
+    }
 
     const osdkObjects = await Promise.all(objectUpdates.map(async (o) => {
+      const keysToDelete = Object.keys(o.object).filter((key) =>
+        sub.requestedReferenceProperties.includes(key)
+      );
+      for (const key of keysToDelete) {
+        delete o.object[key];
```

Comment:
I added this because I wanted the contract here to essentially be if it's undefined, theres no update. Although we do get the property back when it's requested as a reference value, I thought it would be confusing for a user to find this never undefined and there is no latestFetchedValue sometimes, although I don't have a strong opinion here

Reply:
Hmm, if I am showing where a boat is and the backend has timed out the values however it decides to do that and tells me the boat is gone, i think we want it to reflect that?

Reply:
Maybe we use `null` to reflect that we loaded a value but it does not exist?